File: associative_arrays.rb
# 1. initialization
puts "1. initialization -----------------------------------"
hash1 = {"name" => "Melisa", "department" => "CS" , "ID" => 21803668}
puts hash1

# 2. getting value for a key
puts "\n2. getting value for a key --------------------------"
puts hash1["name"]

# 3. adding an element
puts "\n3. adding an element --------------------------------"
puts hash1
hash1["age"] = "21"
puts hash1

# 4. removing an element
puts "\n4. removing an element ------------------------------"
puts hash1
hash1.delete( "department" )
puts hash1

# 5. modifying existing element
puts "\n5. modifying existing element ----------------------"
puts hash1
hash1["age"] = "22"
puts hash1

# 6. searching for a key
puts "\n6. searching for a key ------------------------------"
puts hash1.key? "name"
puts hash1.key? "surname" 

# 7. searching for a value
puts "\n7. searching for a value ----------------------------"
puts hash1.has_value? "Melisa"
puts hash1.has_value? "Yeliz"

# 8. looping
puts "\n8. looping ------------------------------------------"

def foo( key, value ) 
    puts key + ": " + value.to_s
end

hash1.each do |key, value|
  foo( key, value )
end

File: associative_arrays.py
# 1. initialization
print( "1. initialization -----------------------------------")
dict1 = {"name": "Melisa", 
            "department": "CS", 
            "ID": 21803668 }
print( dict1 )

dict2 = {"name": "Melisa1", 
            "name": "Melisa2", 
            "ID": 21803668 }
print(dict2)

# 2. getting value for a key
print( "2. getting value for a key --------------------------")
print( dict1["name"] )

# 3. adding an element
print( "3. adding an element --------------------------------")
print( dict1 )
dict1["age"] = "21"
print( dict1 )

# 4. removing an element
print( "4. removing an element ------------------------------")
print( dict1 )
dict1.pop( "department" )
print( dict1 )

# 5. modifying existing element
print( "5. modifying existing element ----------------------")
print( dict1 )
dict1["age"] = "22"
print( dict1 )

# 6. searching for a key
print( "6. searching for a key ------------------------------")
print( 'name' in dict1 )
print( 'surname' in dict1 )

# 7. searching for a value
print( "7. searching for a value ----------------------------")
values = list( dict1.values() )
print( "Melisa" in values )
print( "Yeliz" in values )

# 8. looping
print( "8. looping ------------------------------------------")

def foo( key, value ):
  print( str(key) + ": " + str(value) )
  
for key, value in dict1.items():
    foo( key, value )

